iroha/src/lib.rs

```diff
@@ -18,31 +18,48 @@ pub mod torii;
 pub mod tx;
 pub mod wsv;
 
-use std::{convert::Infallible, sync::Arc, time::Duration};
+use std::{sync::Arc, time::Duration};
 
 use genesis::GenesisNetworkTrait;
 use iroha_actor::{broker::*, prelude::*};
 use iroha_data_model::prelude::*;
 use iroha_error::{error, Result, WrapErr};
+use parity_scale_codec::{Decode, Encode};
 use smartcontracts::permissions::{IsInstructionAllowedBoxed, IsQueryAllowedBoxed};
 use tokio::{sync::broadcast, task::JoinHandle};
 use wsv::{World, WorldTrait};
 
 use crate::{
     block::VersionedValidBlock,
-    block_sync::{BlockSynchronizer, BlockSynchronizerTrait},
+    block_sync::{
+        message::VersionedMessage as BlockSyncMessage, BlockSynchronizer, BlockSynchronizerTrait,
+    },
     config::Configuration,
     genesis::GenesisNetwork,
     kura::{Kura, KuraTrait},
     prelude::*,
     queue::{Queue, QueueTrait},
-    sumeragi::{Sumeragi, SumeragiTrait},
+    sumeragi::{message::VersionedMessage as SumeragiMessage, Sumeragi, SumeragiTrait},
     torii::Torii,
 };
 
 /// The interval at which sumeragi checks if there are tx in the `queue`.
 pub const TX_RETRIEVAL_INTERVAL: Duration = Duration::from_millis(100);
 
+/// Specialized type of Iroha Network
+pub type IrohaNetwork = iroha_p2p::Network<NetworkMessage>;
+
+/// The network message
+#[derive(Clone, Debug, Encode, Decode, iroha_actor::Message)]
+pub enum NetworkMessage {
+    /// Blockchain message
+    SumeragiMessage(Box<SumeragiMessage>),
+    /// Block sync message
+    BlockSync(Box<BlockSyncMessage>),
```

Comment:
Could you recheck it also, not sure if it should be boxed

Reply:
Checked 3 days ago.

Reply:
I bet problem is with hashes or signatures (as they are 64 bytes long), may be a better idea will be to box them?

Reply:
No, the problem is with messages itself, they may be of arbitrary sizes.

Reply:
No, they all implement `Sized`

Reply:
```
error: enum variant is more than three times larger (344 bytes) than the next largest
  --> iroha\src\lib.rs:56:5
   |
56 |     SumeragiMessage(SumeragiMessage),
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: requested on the command line with `-D variant-size-differences`
```

Reply:
Agreed on merging as it is.